controllers/nstemplateset/cluster_resources_test.go

```diff
@@ -606,297 +353,170 @@ func TestDeleteClusterResources(t *testing.T) {
-				AssertThatNSTemplateSet(t, namespaceName, spacename, cl).
-					HasFinalizer().
-					HasConditions(Provisioning())
-				AssertThatCluster(t, cl).
-					HasResource("for-"+spacename, &quotav1.ClusterResourceQuota{},
-						WithLabel("toolchain.dev.openshift.com/tier", "advanced"),
-						Containing(`"limits.cpu":"2","limits.memory":"10Gi"`)).
-					HasResource(spacename+"-tekton-view", &rbacv1.ClusterRoleBinding{},
-						WithLabel("toolchain.dev.openshift.com/tier", "advanced"))
-			})
-		})
-
-		t.Run("upgrade from team to advanced with enabled features by updating existing CRQ", func(t *testing.T) {
-			// given
 			nsTmplSet := newNSTmplSet(namespaceName,
-				spacename,
+				spaceName,
 				"advanced",
 				withClusterResources("abcde11"),
+				withStatusClusterResourcesInTier("team", "previousrevision"),
 				withNSTemplateSetFeatureAnnotation("feature-1"))
-			devNs := newNamespace("team", spacename, "dev")
-			crq := newClusterResourceQuota(spacename, "team")
-			manager, cl := prepareClusterResourcesManager(t, nsTmplSet, crq, devNs)
+			devNs := newNamespace("team", spaceName, "dev")
+			crq := newClusterResourceQuota(spaceName, "team")
+			crq.Spec.Quota.Hard["limits.cpu"] = resource.MustParse("100m")
+			manager, cl := prepareClusterResourcesManager(t, nsTmplSet, crq, devNs, previousTierTemplate)
 
 			// when
-			updated, err := manager.ensure(ctx, nsTmplSet)
+			err = manager.ensure(ctx, nsTmplSet)
 
 			// then
 			require.NoError(t, err)
-			assert.True(t, updated)
-			AssertThatNSTemplateSet(t, namespaceName, spacename, cl).
+			AssertThatNSTemplateSet(t, namespaceName, spaceName, cl).
 				HasFinalizer().
 				HasConditions(Updating())
 			AssertThatCluster(t, cl).
-				HasResource("for-"+spacename, &quotav1.ClusterResourceQuota{},
-					WithLabel("toolchain.dev.openshift.com/templateref", "advanced-clusterresources-abcde11"),
-					WithLabel("toolchain.dev.openshift.com/tier", "advanced"),
+				HasResource("for-"+spaceName, &quotav1.ClusterResourceQuota{},
 					Containing(`"limits.cpu":"2","limits.memory":"10Gi"`)).
-				HasNoResource("feature-1-for-"+spacename, &quotav1.ClusterResourceQuota{})
-
-			t.Run("upgrade from team to advanced by creating featured CRQ since regular CRQ is already created", func(t *testing.T) {
-				// when
-				updated, err := manager.ensure(ctx, nsTmplSet)
-
-				// then
-				require.NoError(t, err)
-				assert.True(t, updated)
-				AssertThatNSTemplateSet(t, namespaceName, spacename, cl).
-					HasFinalizer().
-					HasConditions(Updating())
-				AssertThatCluster(t, cl).
-					HasResource("for-"+spacename, &quotav1.ClusterResourceQuota{}).
-					HasResource("feature-1-for-"+spacename, &quotav1.ClusterResourceQuota{})
-			})
+				HasResource("feature-1-for-"+spaceName, &quotav1.ClusterResourceQuota{})
 		})
 
-		t.Run("downgrade from advanced to team with featured CRQ to be deleted", func(t *testing.T) {
+		t.Run("promote to a new tier with feature enabled", func(t *testing.T) {
```

Comment:
how is this test different to the previous one?
```
"promote to a new tier with a new feature enabled"
```
is it because in this test there is no feature object to be applied in the given tier?

Reply:
Looking at this again, I think the two tests test quite different things:
   1. checks that switching to another tier and simultaneous enabling of a feature works
   1. checks that when switching to a different tier, all the previously deployed features are removed as well.

IMHO it makes sense to test it like this, because it is distinct from just enabling/disabling features and just changing tiers.